pypeit/reduce.py

```diff
@@ -190,64 +143,213 @@ def parse_manual_dict(self, manual_dict, neg=False):
         # Return
         return manual_extract_dict
 
-    def find_objects(self, image, std=False, ir_redux=False, std_trace=None, maskslits=None,
-                     show_peaks=False, show_fits=False, show_trace=False, show=False,
-                     manual_extract_dict=None, debug=False):
+    def extract(self, global_sky, sobjs_obj):
         """
+        Main method to extract spectra from the ScienceImage
 
         Args:
-            image:
-            std:
-            ir_redux:
-            std_trace:
-            maskslits:
-            show_peaks:
-            show_fits:
-            show_trace:
-            show:
-            manual_extract_dict:
-            debug:
+            global_sky (ndarray):
+                Sky estimate
+            sobjs_obj (SpecObjs):
+                List of SpecObj that have been found and traced
+        """
+        # This holds the objects, pre-extraction
+        self.sobjs_obj = sobjs_obj
+
+        # Boxcar only??
+        if self.par['scienceimage']['extraction']['skip_optimal']:
+            msgs.info("Skipping optimal extraction")
+
+            # This will hold the extracted objects
+            self.sobjs = self.sobjs_obj.copy()
+
+            # Quick loop over the objects
+            for iord in range(self.nobj):
+                sobj = self.get_sobj(self.sobjs, iord)
+                plate_scale = self.get_platescale(iord)
+                # True  = Good, False = Bad for inmask
+                thismask = (self.slitmask == iord)  # pixels for this slit
+                inmask = (self.sciImg.mask == 0) & thismask
+                # Do it
+                extract.extract_boxcar(self.sciImg.image, self.sciImg.ivar,
+                                               inmask, self.caliBrate.mswave,
+                                               global_sky, self.sciImg.rn2img,
+                                               self.par['scienceimage']['extraction']['boxcar_radius']/plate_scale,
+                                               sobj)
+            # Fill up extra bits and pieces
+            self.objmodel = np.zeros_like(self.sciImg.image)
+            self.ivarmodel = np.copy(self.sciImg.ivar)
+            self.outmask = self.sciImg.mask
+            self.skymodel = global_sky.copy()
+        else:  # Full extraction
```

Comment:
CHange comment to full extraction: Local Sky subtraction and optimal extraction.

Reply:
done.